rustsec/src/advisory/license.rs

```diff
@@ -0,0 +1,94 @@
+use crate::Error;
+use serde::{de, Deserialize, Deserializer, Serialize, Serializer};
+use std::str::FromStr;
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+/// Type representing allowed licenses for advisory content
+enum LicenseVariant {
+    /// CC0-1.0
+    CreativeCommonsZero10,
+    /// CC-BY-4.0
+    CreativeCommonsCc40,
+    /// Other SPDX requirement
+    Other(String),
+}
```

Comment:
Documenting the SPDX identifier to which it serializes is a good idea.

I dread writing out `CreativeCommonsCc40` every time I need it. Can we put the full license name into the comment and an SPDX-like shorthand as the variant name? I'm fine with `CcBy40` if that's what it takes.

Reply:
Works for me.